TRex2-lib/src/Packets/RulePkt.cc

```diff
@@ -532,30 +489,31 @@ bool RulePkt::operator==(const RulePkt& pkt) const {
   return ruleId == pkt.ruleId;
 }
 
-bool RulePkt::operator!=(const RulePkt& pkt) const {
-  return ruleId != pkt.ruleId;
-}
+bool RulePkt::operator!=(const RulePkt& pkt) const { return !(*this == pkt); }
 
 bool RulePkt::addNegation(int eventType, Constraint* constr, int constrLen,
-                          int lowerId, TimeMs& lowerTime, int upperId) {
+                          int lowerId, const TimeMs& lowerTime, int upperId) {
   int numPredicates = predicates.size();
-  if (lowerId > numPredicates)
+  if (lowerId >= numPredicates) {
```

Comment:
BUGFIX: Change the boundaries check to match the usual `[0, size)` range, is it correct? (See also all the other lines where the check is made)